### PR TITLE
docs: clarify use cases for `pg_basebackup` bootstrap

### LINKED ISSUE #5719: [Docs]: Clarify use cases for `pg_basebackup`
### Is there an existing issue already for your request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new documentation enhancement to be evaluated.

### What problem in the existing documentation this issue aims to solve?

The ["Bootstrap from a live cluster"](https://cloudnative-pg.io/documentation/current/bootstrap/#bootstrap-from-another-cluster) section can be misleading, particularly in the statement:

> The primary use case for this method is represented by migrations to CloudNativePG, either from outside Kubernetes or within Kubernetes (e.g., from another operator).

This claim is, in my opinion, overstated. In practice, the "Requirements" section is often overlooked, misunderstood, or not followed, which can lead to challenges. While theoretically possible, successfully importing a database from outside Kubernetes or from another operator via physical replication is quite *rare*. The process requires many complex factors to align perfectly.

It's worth noting that this section was written when CloudNativePG, at the time a closed-source product within EDB, lacked the import functionality and logical replication was not yet feasible.

### Describe what additions need to be done to the documentation

We should rephrase that sentence and clearly scope down the feature.

### Describe what pages need to change in the documentation, if any

https://cloudnative-pg.io/documentation/current/bootstrap/#bootstrap-from-another-cluster

### Describe what pages need to be removed from the documentation, if any

_No response_

### Additional context

_No response_

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gbartolini**
Closes #5719

### CODE REVIEW

**Review by leonardoce — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

File: docs/src/bootstrap.md [hunk 1]
```diff
@@ -389,44 +389,59 @@ to the ["Recovery" section](recovery.md).
...
-
-Similar to the case of the `recovery` bootstrap method, once the clone operation
-completes, the operator will take ownership of the target cluster, starting from
-the first instance. This includes overriding some configuration parameters, as
-required by CloudNativePG, resetting the superuser password, creating
-the `streaming_replica` user, managing the replicas, and so on. The resulting
-cluster will be completely independent of the source instance.
+    The current implementation creates a *snapshot* of the source PostgreSQL
+    instance when the cloning process finishes, and immediately starts the new
+    cluster. Refer to ["Current limitations"](#current-limitations) for more
```
Comment (jsilvela) [inline docs/src/bootstrap.md:422]: I got a bit confused reading this paragraph.
I parsed it initially as "when the cloning process finishes, we create a snapshot of the source".


```suggestion
    In its current implementation, this method clones the source PostgreSQL
    instance, thereby creating a *snapshot*. Once the cloning process has finished,
    the new cluster is immediately started. Refer to ["Current limitations"](#current-limitations) for more
```

**Review by jsilvela — APPROVED** [review]
Just the one optional comment I left.
In a previous job I remember design documents had a set of goals and non-goals.
Perhaps non-goals is the right expression for this kind of caveat? (also thinking about some of the caveats around declarative databases)

**Comment by jsilvela** [comment]
Looking at the lists of admonitions available on MkDocs, badly missing one: "At your own risk".

**Pull request merged**